freeipa/src/main/java/com/sequenceiq/freeipa/client/FreeIpaClientBuilder.java

```diff
@@ -66,43 +67,66 @@
 
     private final SSLContext sslContext;
 
-    private final String port;
+    private final int port;
 
     private final HttpClientConfig clientConfig;
 
-    public FreeIpaClientBuilder(String user, String pass, String realm, HttpClientConfig clientConfig, String port) throws Exception {
+    private Map<String, String> additionalHeaders;
+
+    public FreeIpaClientBuilder(String user, String pass, String realm, HttpClientConfig clientConfig, int port) throws Exception {
         this.user = user;
         this.pass = pass;
         this.realm = realm;
         this.clientConfig = clientConfig;
         this.port = port;
-        sslContext = setupSSLContext(clientConfig.getClientCert(), clientConfig.getClientKey(), clientConfig.getServerCert());
-        RegistryBuilder<ConnectionSocketFactory> registryBuilder = RegistryBuilder.create();
-        SSLConnectionSocketFactory socketFactory = new SSLConnectionSocketFactory(sslContext, hostnameVerifier());
-        registryBuilder.register("https", socketFactory);
-        Registry<ConnectionSocketFactory> registry = registryBuilder.build();
-        connectionManager = new PoolingHttpClientConnectionManager(registry);
+        additionalHeaders = Map.of();
+
+        if (clientConfig.hasSSLConfigs()) {
+            this.sslContext =
+                setupSSLContext(clientConfig.getClientCert(), clientConfig.getClientKey(), clientConfig.getServerCert());
+            RegistryBuilder<ConnectionSocketFactory> registryBuilder = RegistryBuilder.create();
+            SSLConnectionSocketFactory socketFactory = new SSLConnectionSocketFactory(sslContext, hostnameVerifier());
+            registryBuilder.register("https", socketFactory);
+            Registry<ConnectionSocketFactory> registry = registryBuilder.build();
+            connectionManager = new PoolingHttpClientConnectionManager(registry);
+        } else {
+            this.sslContext = null;
+            connectionManager = new PoolingHttpClientConnectionManager();
+        }
+
         connectionManager.setMaxTotal(CONNECTION_POOL_MAX);
         connectionManager.setDefaultMaxPerRoute(CONNECTION_POOL_MAX_PER_ROOT);
     }
 
-    public FreeIpaClientBuilder(String user, String pass, String realm, HttpClientConfig clientConfig, String port, String basePath) throws Exception {
+    public FreeIpaClientBuilder(String user, String pass, String realm,
+                                HttpClientConfig clientConfig, int port,
+                                String basePath, Map<String, String> additionalHeaders) throws Exception {
         this(user, pass, realm, clientConfig, port);
```

Comment:
I think it makes sense to flip the constructor delegation around so that `FreeIpaClientBuilder(String user, String pass, String realm, HttpClientConfig clientConfig, int port)` calls `this(user, pass, realm, clientConfig, port, DEFAULT_BASE_PATH, ImmutableMap.of())` instead.